src/cdk/v2/destinations/intercom/procWorkflow.yaml

```diff
@@ -0,0 +1,170 @@
+bindings:
+  - name: EventType
+    path: ../../../../constants
+  - path: ./utils
+    exportAll: true
+  - path: ../../bindings/jsontemplate
+    exportAll: true
+  - name: defaultRequestConfig
+    path: ../../../../v0/util
+  - name: removeUndefinedAndNullValues
+    path: ../../../../v0/util
+  - name: getFieldValueFromMessage
+    path: ../../../../v0/util
+  - name: isDefinedAndNotNull
+    path: ../../../../v0/util
+  - name: addExternalIdToTraits
+    path: ../../../../v0/util
+
+steps:
+  - name: checkIfProcessed
+    condition: .message.statusCode
+    template: |
+      $.batchMode ? .message.body.JSON : .message;
+    onComplete: return
+
+  - name: messageType
+    template: |
+      .message.type.toLowerCase();
+
+  - name: validateInput
+    template: |
+      let messageType = $.outputs.messageType;
+      $.assert(messageType, "message Type is not present. Aborting");
+      $.assert(messageType in {{$.EventType.([.IDENTIFY, .TRACK, .GROUP])}}, "message type " + messageType + " is not supported");
+      $.assertConfig(.destination.Config.apiKey, "Access Token is not present. Aborting");
+
+  - name: rEtlPayload
+    condition: .message.context.mappedToDestination === true
+    template: |
+      $.addExternalIdToTraits(.message);
+      const payload = $.getFieldValueFromMessage(.message, "traits");
+      payload;
+
+  - name: searchContact
+    condition: $.outputs.messageType === {{$.EventType.IDENTIFY}} || $.outputs.messageType === {{$.EventType.GROUP}}
+    template: |
+      const contactId = await $.searchContact(.message, .destination);
+      contactId;
+
+  - name: identifyEtlPayload
+    condition: $.outputs.messageType === {{$.EventType.IDENTIFY}} && !.message.context.mappedToDestination
+    template: |
+      const payload = .message.({
+        external_id: {{{{$.getGenericPaths("userIdOnly")}}}},
+        email: {{{{$.getGenericPaths("email")}}}},
+        phone: {{{{$.getGenericPaths("phone")}}}},
+        avatar: {{{{$.getGenericPaths("avatar")}}}},
+        last_seen_at: $.toSeconds(.context.traits.lastSeenAt),
+        role: .traits.role || .context.traits.role,
+        signed_up_at: $.toSeconds(.traits.createdAt || .context.traits.createdAt),
+        owner_id: .traits.ownerId || .context.traits.ownerId ? Number(.traits.ownerId || .context.traits.ownerId) : undefined,
+        unsubscribed_from_emails: .traits.unsubscribedFromEmails || .context.traits.unsubscribedFromEmails
+      })
+      payload.name = $.getName(.message);
+      payload.custom_attributes = .message.context.traits || {};
+      payload.custom_attributes = $.filterCustomAttributes(payload, "user");
+      payload.external_id = !(payload.external_id) && .destination.Config.sendAnonymousId && .message.anonymousId ? .message.anonymousId : payload.external_id;
```

Comment:
no it will not give same result. batter to keep it as 
`payload.external_id = !payload.external_id && .destination.Config.sendAnonymousId && .message.anonymousId ? .message.anonymousId : payload.external_id;`

Reply:
which case these 2 are different

Reply:
`.destination.Config.sendAnonymousId && .message.anonymousId` this will be either true or false it will not assign any value